code/modules/mob/living/living.dm

```diff
@@ -546,67 +558,29 @@ Sorry Giacom. Please don't be mad :(
 		else
 			return 0
 
-	if (restrained())
+	var/atom/movable/pullee = pulling
+	if(pullee && get_dist(src, pullee) > 1)
 		stop_pulling()
-
-
-	var/cuff_dragged = 0
-	if (restrained())
-		for(var/mob/living/M in range(1, src))
-			if (M.pulling == src && !M.incapacitated())
-				cuff_dragged = 1
```

Comment:
We no longer check everything around us in a one tile radius for a mob pulling us every time we move while restrained.
With the pulledby var we can handle this in Client/Move()'s Process_Grab(). And we don't need to check if our puller is incapacitated, since every thing that can incapacitate a mob now automatically stops its pull, so all we have to check is if we're being pulled.